apollo-router/src/services/execution_service.rs

```diff
@@ -128,32 +144,36 @@ where
                                 sub_responses.push((path.clone(), value.clone()));
                             });
 
-                            Response::builder()
-                                .has_next(has_next)
-                                .incremental(
-                                    sub_responses
-                                        .into_iter()
-                                        .map(move |(path, data)| {
-                                            let errors = response
-                                                .errors
-                                                .iter()
-                                                .filter(|error| match &error.path {
-                                                    None => false,
-                                                    Some(err_path) => err_path.starts_with(&path),
-                                                })
-                                                .cloned()
-                                                .collect::<Vec<_>>();
-                                            IncrementalResponse::builder()
-                                                .and_label(response.label.clone())
-                                                .data(data)
-                                                .path(path)
-                                                .errors(errors)
-                                                .extensions(response.extensions.clone())
-                                                .build()
-                                        })
-                                        .collect(),
-                                )
-                                .build()
+                            ready(Some(
+                                Response::builder()
+                                    .has_next(has_next)
+                                    .incremental(
+                                        sub_responses
+                                            .into_iter()
+                                            .map(move |(path, data)| {
+                                                let errors = response
+                                                    .errors
```

Comment:
NIT: I think you can `drain(..)` it to not clone errors.

Reply:
I can't, because we go through the list of errors for each sub response (iterating on line 152)